### PR TITLE
Add workflow to update cspell.json

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Adds GitHub Actions workflow to update cspell.json.

The workflow is triggered below two times:

- `workflow_dispatch` (manually)
- `schedule` (once a month)

The workflow creates a Pull Request like https://github.com/sosukesuzuki/prettier/pull/504

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: .github/workflows/update-cspell-json.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            const dateStr =
+              date.getFullYear() +
+              "-" +
+              ("0" + (date.getMonth() + 1)).slice(-2) +
+              "-" +
+              ("0" + date.getDate()).slice(-2) +
+              "-" +
+              ("0" + date.getHours()).slice(-2) +
+              "-" +
+              ("0" + date.getMinutes()).slice(-2);
```
Comment (fisker) [inline .github/workflows/update-cspell-json.yml:46]:
```suggestion
            const dateStr =new Date().toJSON().replace(/\D/g, '-').slice(0, 16);
```

Comment (fisker) [inline .github/workflows/update-cspell-json.yml:46]:
```suggestion
            const dateStr = new Date().toJSON().replace(/\D/g, "-").slice(0, 16);
```

File: .github/workflows/update-cspell-json.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            const branch = `update-cspell-${dateStr}`;
+
+            await execa("git", ["checkout", "-b", branch]);
+
+            console.log("Run ./scripts/clean-cspell.mjs");
+            await execa("node", ["./scripts/clean-cspell.mjs"]);
+
+            const { stdout: diff } = await execa("git", ["diff", "--name-only"]);
+            if (diff.includes("cspell.json")) {
+              await execa("git", ["add", "."]);
```
Comment (fisker) [inline .github/workflows/update-cspell-json.yml:56]:
```suggestion
              await execa("git", ["add", "cspell.json"]);
```

File: .github/workflows/update-cspell-json.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+              await execa("git", ["commit", "-m", "Update clean-cspell.json"]);
```
Comment (fisker) [inline .github/workflows/update-cspell-json.yml:57]:
```suggestion
              await execa("git", ["commit", "--message", "Update cspell.json"]);
```

File: .github/workflows/update-cspell-json.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+              console.log("Push to remote...");
+              await execa("git", ["push", "origin", branch]);
+
+              console.log("Create a pull request...");
+              await github.pulls.create({
+                ...context.repo,
+                title: `Update cspell.json (${dateStr})`,
```
Comment (fisker) [inline .github/workflows/update-cspell-json.yml:65]:
```suggestion
                title: `Update cspell.json (${new Date().toJSON().slice(0, 10)})`,
```

File: .github/workflows/update-cspell-json.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+name: "Update cspell.json"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "00 01 1 * *"
```
Comment (fisker) [inline .github/workflows/update-cspell-json.yml:6]: Is it possible to check repo is not fork? Otherwise, PR will be created on forks.

**Review by fisker — CHANGES_REQUESTED** [review]
I'm not sure if this is necessary, but few comments on code.

**Comment by sosukesuzuki** [comment]
Thanks for the review. I'll fix it later.
Btw, do you think this workflow is necessary? Personally, I find it tedious to run the script locally, so the maintainers sometimes forget about it. So I thought that an automatically triggered workflow would prevent that from happening.

**Comment by fisker** [comment]
> do you think this workflow is necessary? 

I don't think it's necessary.

> I find it tedious to run the script locally

The dictionary can have unused words, for example, we added `foo`, but removed later, we can keep it in the dictionary, because another contributor may use it later.


I say we can run the script yearly, or never.... Added that script just in case someone want to clean.

**Comment by sosukesuzuki** [comment]
Okay, I'm good to hear from you who create the script. Let's close this.

**Pull request closed without merging**